lib/stat_tracker.rb

```diff
@@ -198,6 +199,16 @@ def percent_wins_by_coach(season_id)
     end
   end
 
+  def num_games_per_opponent_for_team(team_id)
+    opponent_by_game_id_for_team(team_id).each_with_object(Hash.new(0)) do |(game_id, team_id), result| 
+      result[team_id] += 1 if team_id == team_id 
+    end 
+  end
```

Comment:
Nice! Is each_with_object like a better reduce for hashes?

Reply:
I had been seeing some arguments online that each_with_object could be a better choice for hash transformations. I originally tried with .reduce but since I'm running the method on a hash, each_with_object seemed cleaner? Totally a matter of opinion I think but I do like how it doesn't reassign each time through and I don't have return result each time.